src/metabase/native_query_analyzer/replacement.clj

```diff
@@ -22,6 +25,10 @@
   [raw-query]
   (first-unique raw-query #(apply format "(%s = %s)" (repeat 2 (gensymed-string)))))
 
+(defn- sentinel-table
```

Comment:
nit: I think its worth highlighting that this function is impure and generates something unique. Functions with raw nouns as names should typically return deterministic values in my option.

Also it's hard to articulate why, but I feel it's more natural to use `sentinel` as the noun and `table` as the adjective. We have a bunch of different types of sentinel tokens in the same way that we have different types of clauses. A select clause, a table sentinel.

```suggestion
(defn- gen-table-sentinel
```

Reply:
yeah, wfm, fixed